code/entities/cooking/hotdog-roller/components/interactions/switches/BaseSwitch.cs

```diff
@@ -0,0 +1,43 @@
+﻿using Sandbox;
+
+namespace Cinema;
+
+public partial class BaseSwitch : EntityComponent<HotdogRoller>
```

Comment:
You could refactor `BaseSwitch` `LeftSwitch` and `RightSwitch` into a single class, `HotdogRollerSwitch`

The only difference between these classes is `LeftSwitch` sets the parameter (toggle_left) and `RightSwitch` sets the parameter (toggle_right)

Reply:
That's definitely something I want to rethink / refactor. I originally had it as one class, but the way the model is setup currently, I found it to be cleaner to have two separate classes for now